crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -189,6 +189,58 @@ impl<'a> From<&'a EntityMut<'_>> for EntityRef<'a> {
     }
 }
 
+impl<'a> TryFrom<FilteredEntityRef<'a>> for EntityRef<'a> {
+    type Error = ();
```

Comment:
Does worth it adding an error for this conversion, so users may know why it failed, something like:
```rust
pub enum EntityRefAccessError {
    NoReadAll,
    NoWriteAll,
}

impl fmt::Display for EntityRefAccessError {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match self {
            Self::NoReadAll=> write!(f, "The given FilteredEntityRef has no read access on all entities."),
            Self::NoWriteAll=> write!(f, "The given FilteredEntityRef has no write access on all entities."),
        }
    }
}

impl std::error::Error for IdentifierError {}
```

Reply:
Added the error enum, went with slightly different names to match the naming conventions for existing `TryFrom` errors, and also for variants of [this existing error type](crates/bevy_ecs/src/query/error.rs)